File: Easy/Javascript/1137. N-th Tribonacci Number.js
/**The Tribonacci sequence Tn is defined as follows: 

T0 = 0, T1 = 1, T2 = 1, and Tn+3 = Tn + Tn+1 + Tn+2 for n >= 0.

Given n, return the value of Tn. */

/**
 * @param {number} n
 * @return {number}
 */
var tribonacci = function(n) {
    let dp = [0, 1, 1];

    for (let i = 3; i <= n; i++){
        dp[i] = dp[i-1] + dp[i - 2] + dp[i - 3]
    }

    return dp[n]
};

File: Easy/Python/1137. N-th Tribonacci Number.py
"""
The Tribonacci sequence Tn is defined as follows: 

T0 = 0, T1 = 1, T2 = 1, and Tn+3 = Tn + Tn+1 + Tn+2 for n >= 0.

Given n, return the value of Tn.
"""

class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0:
            return 0
        if n == 1 or n == 2:
            return 1

        t = 0
        t1 = 1
        t2 = 1
        tn = None

        for i in range(3,n+1):
            tn = t + t1 + t2
            t = t1
            t1 = t2
            t2 = tn

        return tn